src/olympia/devhub/views.py

```diff
@@ -2130,44 +2130,38 @@ def email_verification(request):
         return redirect('devhub.email_verification')
 
     if email_verification:
-        if not email_verification.is_expired:
-            if (
-                email_verification.status
-                == SuppressedEmailVerification.STATUS_CHOICES.Pending
-            ):
-                if email_verification.is_timedout:
-                    data['state'] = VERIFY_EMAIL_STATE['verification_timedout']
-                else:
-                    data['state'] = VERIFY_EMAIL_STATE['verification_pending']
-            elif (
-                email_verification.status
-                == SuppressedEmailVerification.STATUS_CHOICES.Failed
-            ):
-                data['state'] = VERIFY_EMAIL_STATE['verification_failed']
-            elif (
-                email_verification.status
-                == SuppressedEmailVerification.STATUS_CHOICES.Delivered
-            ):
-                if code := request.GET.get('code'):
-                    if code == email_verification.confirmation_code:
-                        suppressed_email.delete()
-                        send_mail_jinja(
-                            gettext('Your email has been verified'),
-                            'devhub/emails/verify-email-completed.ltxt',
-                            {},
-                            recipient_list=[request.user.email],
-                        )
-
-                    if SuppressedEmailVerification.objects.filter(
-                        confirmation_code=code
-                    ).exists():
-                        data['state'] = VERIFY_EMAIL_STATE['confirmation_unauthorized']
-                    else:
-                        return redirect('devhub.email_verification')
-                else:
-                    data['state'] = VERIFY_EMAIL_STATE['confirmation_pending']
-        else:
+        if email_verification.is_expired:
             data['state'] = VERIFY_EMAIL_STATE['verification_expired']
+        elif code := request.GET.get('code'):
```

Comment:
@eviljeff this is the change. I've extracted the check for a code the URL to be independent of the status of the email verification.

Before: we only check for a valid code if we confirm the email verification is sent by socketlbabs

Now: if a valid code is present then we validate it regardless of the status.